pkg/arvo/sys/arvo.hoon

```diff
@@ -1,85 +1,177 @@
-::::::  ::::::::::::::::::::::::::::::::::::::::::::::::::::::
-::::::  ::::::    volume 3, Arvo models and skeleton    ::::::
-::::::  ::::::::::::::::::::::::::::::::::::::::::::::::::::::
+=<
+::  arvo formal interface
+::
+::    this lifecycle wrapper makes the arvo door look like a gate to fit
+::    urbit's formal lifecycle function. a practical interpreter
+::    can ignore it.
+::
+|=  [now=@da ovo=*]
+^-  *
+~>  %slog.[0 leaf+"arvo-event"]
+.(+> +:(poke now ovo))
+::
 =>
+::  arvo models and skeleton
+::
 |%
-++  arch  {fil/(unit @uvI) dir/(map @ta $~)}            ::  fundamental node
-++  arvo  (wind {p/term q/mill} mill)                   ::  arvo card
-++  beam  {{p/ship q/desk r/case} s/path}               ::  global name
-++  beak  {p/ship q/desk r/case}                        ::  garnish with beak
-++  bone  @ud                                           ::  opaque duct
-++  case                                                ::  version
-          $%  {$da p/@da}                               ::  date
-              {$tas p/@tas}                             ::  label
-              {$ud p/@ud}                               ::  sequence
-          ==                                            ::
-++  desk  @tas                                          ::  ship desk case spur
-++  dock  (pair @p term)                                ::  message target
-++  cage  (cask vase)                                   ::  global metadata
-++  cask  |*(a/mold (pair mark a))                      ::  global data
-++  curd  {p/@tas q/*}                                  ::  typeless card
-++  duct  (list wire)                                   ::  causal history
-++  hypo  |*(a/mold (pair type a))                      ::  type associated
-++  hobo  |*  a/mold                                    ::  kiss wrapper
-          $?  $%  {$soft p/*}                           ::
-              ==                                        ::
-              a                                         ::
-          ==                                            ::
-++  mark  @tas                                          ::  content type
-++  mash  |=(* (mass +<))                               ::  producing mass
-++  mass  $~  [%$ [%& ~]]                               ::  memory usage
-          (pair cord (each noun (list mash)))           ::
-++  mill  (each vase milt)                              ::  vase+metavase
-++  milt  {p/* q/*}                                     ::  metavase
-++  monk  (each ship {p/@tas q/@ta})                    ::  general identity
-++  muse  {p/@tas q/duct r/arvo s/@ud}                  ::  sourced move
-++  move  {p/duct q/arvo}                               ::  arvo move
-++  ovum  {p/wire q/curd}                               ::  typeless ovum
-++  pane  (list {p/@tas q/vase})                        ::  kernel modules
-++  pone  (list {p/@tas q/vise})                        ::  kernel modules old
+::  +arch: fundamental node
```

Comment:
It seems to me that the long-form comment style is helpful because it allows longer descriptions, not for its own sake.  The previous version of this is significantly more readable to me (comment-wise; the syntax modernizations are righteous).

Reply:
I do think all of these things deserve longer-form commentary.  E.g. `+arch` -- "fundamental node" is uninformative, to me, whereas some longer exposition on it being a blob ref / directory listing would be more appropriate.  This struck me as moving partway towards that goal.

However, maybe it'd be better to add more useful comments first, *then* modernise their style..

Reply:
Having ruminated on this a little more: I do think the breathing comment style is an improvement, even absent any additional meat in the actual comments.  But it's certainly a controversial change, so I'll revert it for the time being.